distributed/deploy/local.py

```diff
@@ -100,6 +110,7 @@ class LocalCluster(SpecCluster):
     def __init__(
         self,
         name=None,
+        memory_limit=None,
```

Comment:
I don't think we want to add this parameter to `LocalCluster`. The variable `memory_limit` isn't used anywhere else in the body of this function. So by adding it here, it's "sucking up" the `memory_limit` argument and not passing it on to the worker, so in the end, the memory limit doesn't actually get applied.

Previously, when you called `LocalCluster(..., memory_limit=...)`, the `memory_limit` argument would end up in `**worker_kwargs`. That kwargs dict would end up here: https://github.com/dask/distributed/blob/3647cfefd004281dac0ffa92349e23bc2a94d68a/distributed/deploy/local.py#L235

Through a chain of calls, those `worker_kwargs`, including `memory_limit`, would eventually be passed into the `Worker` class. The `LocalCluster` doesn't actually use the argument directly.

You can see how some tests are failing indicating that the `memory_limit` wasn't passed down: https://github.com/dask/distributed/runs/7846945041?check_suite_focus=true#step:11:1915

Reply:
Thank you for the explanation!